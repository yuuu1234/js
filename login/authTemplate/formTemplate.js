import React from 'react';
import {
  SafeAreaView, KeyboardAvoidingView, Platform, ScrollView,
} from 'react-native';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import styles from './style';

const withAuthTemplate = ({ Header, Form, Footer }) => {
  @observer
  class Screen extends React.Component {
    handleNavigation = (screenType) => {
      const { navigation } = this.props;
      navigation.navigate(screenType);
    }

    render() {
      return (
        <SafeAreaView style={styles.container}>
          <KeyboardAvoidingView
            style={styles.container}
            behavior={Platform.OS === 'ios' ? 'padding' : null}
          >
            <ScrollView
              showsVerticalScrollIndicator={false}
              keyboardShouldPersistTaps="handled"
              contentContainerStyle={{ flexGrow: 1 }}
            >
              {Header && <Header />}
              {Form && <Form handleNavigation={this.handleNavigation} />}
            </ScrollView>
            {Footer && <Footer handleNavigation={this.handleNavigation} />}
          </KeyboardAvoidingView>
        </SafeAreaView>
      );
    }
  }

  Screen.propTypes = {
    navigation: PropTypes.shape({ navigate: PropTypes.func }).isRequired,
  };

  return Screen;
};

export { withAuthTemplate };
