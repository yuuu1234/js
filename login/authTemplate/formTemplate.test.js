import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import PropTypes from 'prop-types';
import { withAuthTemplate } from './formTemplate';

const MockHeader = () => <div testId="mock-header">Header</div>;
const MockForm = ({ handleNavigation }) => <div handleNavigation={handleNavigation} testId="mock-form">Form</div>;
const MockFooter = ({ handleNavigation }) => <div handleNavigation={handleNavigation} testId="mock-footer">Footer</div>;

MockForm.propTypes = { handleNavigation: PropTypes.func.isRequired };
MockFooter.propTypes = { handleNavigation: PropTypes.func.isRequired };

describe('withAuthTemplate', () => {
  let Platform;

  const params = {
    Header: MockHeader,
    Form: MockForm,
    Footer: MockFooter,
  };
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };
  const WithAuthComponent = withAuthTemplate(params);
  const component = (
    <WithAuthComponent {...props} />
  );
  let wrapper;

  beforeEach(() => {
    Platform = require('react-native').Platform;
    wrapper = mount(component);
  });

  it('should render component correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call navigate correctly', () => {
    const form = wrapper.find({ testId: 'mock-form' });
    form.props().handleNavigation('mock screen');

    expect(props.navigation.navigate).toHaveBeenCalledWith('mock screen');
  });

  it('should set behavior to null if platform is not ios', () => {
    Platform.OS = 'android';
    wrapper = mount(component);
    const keyboard = wrapper.find('KeyboardAvoidingView');
    expect(keyboard.props().behavior).toEqual(null);
  });
});
