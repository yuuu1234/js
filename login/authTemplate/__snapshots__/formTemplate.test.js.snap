// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuthTemplate should render component correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="handled"
      showsVerticalScrollIndicator={false}
    >
      <View>
        <div
          testId="mock-header"
        >
          Header
        </div>
        <div
          handleNavigation={[Function]}
          testId="mock-form"
        >
          Form
        </div>
      </View>
    </RCTScrollView>
    <div
      handleNavigation={[Function]}
      testId="mock-footer"
    >
      Footer
    </div>
  </View>
</RCTSafeAreaView>
`;
